src/ops/kernels/element_binary_kernels.cu

```diff
@@ -266,29 +266,68 @@ void forward_kernel(ElementBinaryMeta const *m,
   // cudnnOpTensor
   if (m->broadcast_input1) {
     // currently only handle add and sub
-    assert(m->op_type == OP_EW_SUB || m->op_type == OP_EW_ADD);
-    checkCUDNN(cudnnOpTensor(m->handle.dnn,
-                             m->opDesc,
-                             &beta,
-                             m->outputTensor,
-                             out_ptr,
-                             &alpha1,
-                             m->input1Tensor,
-                             in1_ptr,
-                             &beta,
-                             m->outputTensor,
-                             out_ptr));
-    checkCUDNN(cudnnOpTensor(m->handle.dnn,
-                             m->opDesc,
-                             &beta,
-                             m->outputTensor,
-                             out_ptr,
-                             &alpha2,
-                             m->input2Tensor,
-                             in2_ptr,
-                             &alpha1,
-                             m->outputTensor,
-                             out_ptr));
+    assert(m->op_type == OP_EW_SUB || m->op_type == OP_EW_ADD ||
+           m->op_type == OP_EW_MUL);
+    if(m->op_type == OP_EW_SUB || m->op_type == OP_EW_ADD){
+      // output = (beta*output + alpha1*input1) + beta*output = input1
```

Comment:
Are these comments useful? If not, will remove then

Reply:
> @jiazhihao I am trying to implement broadcasting for elementwise multiplication. I think I've got the forward pass. I had some questions about the backward pass. I think the computation will be as:
> 
> ```
> in1_grad = in1_grad + reduce(out_grad * in2)
> and similarly for in2_grad
> ```
> 
> Which I think would require storing `out_grad * in2` in a temporary variable. And maybe another one for `in2_grad` due to the different shapes. But, these two temporary variables would not be used for the other elementwise binary operations and probably cause some memory overhead.
> 
> Would there be some simpler method to implement it without requiring so many changes? Or will we have to add in two temporary variables?

The backward equation you showed is correct, and I think you are right that we need a temporary buffer for saving out_grad * in2. An easy way to create it is use part of workspace. Note that workspace is allocated for operators that require saving intermediate buffers (e.g., convolutions, and this one). You can reserve part of the workspace for saving  `out_grad * in2` and the remaining for `cudnnReduceTensor`, which also requires a workspace (https://docs.nvidia.com/deeplearning/cudnn/api/index.html#cudnnReduceTensor)